src/Stratis.Bitcoin.Tests/Connection/ConnectionManagerTest.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Configuration.Settings;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Tests.Common;
+using Stratis.Bitcoin.Tests.Common.Logging;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.Connection
```

Comment:
Add tests for:
- IPv6 localhost and network address
  **Example:** `::1`, `fe80::d111:a4c4:ce4:2bc7%21`

- Any whitebind IP can be mapped to any localhost IPV4/IPV6 address (e.g. `127.0.0.0/8` and `::1/128`)
   `0.0.0.0:16178` == `127.0.0.1:16178`
   `0.0.0.0:16178` == `127.0.0.2:16178`
   `::/128` == `::1`

- If the address could be matched but the port is not matching the method should return false
   `0.0.0.0:16178` != `127.0.0.1:999`
   `::/128:16178` == `::1:16178`

- If we have bound to all interfaces but whitebind only whitelists one interface
   `127.0.0.1:16178` != `0.0.0.0:16178`
   `::1:16178` !=  `::/128:16178`

- Specify only IP address and use the default port

Reply:
Tests are not covering the code from `ConnectionManagerSettings.cs`

Reply:
Personally I would prefer a single test with a list of addresses to compare and expected results.